Automattic/simplenote-ios
# Navigation Bar: Drops custom metrics

### Details:
In this PR we're dropping all of the customized Navigation Bar metrics. All of them!

@eshurakov Sir!! May I bug you with this one?  (Thaaank youuu!)


@SylvesterWilmott Having all of the default OS animations within the Navigation Title feels **so perfect!!**

Closes #918
Closes #852

### Test: List
1. Access the Notes List

- [x] Verify on the top left corner there's a navigation button that allows you to access the Sidebar
- [x] Verify on the top right corner there's a navigation button that allows you to add a new Note

### Test: Editor
1. Access the Notes List
2. Press over any note

- [x] Verify the top left navigation button allows you to go back to the Notes List
- [x] Verify that the `( i )` navigation button presents the Note Options
- [x] Verify that the `+` navigation button adds a new Note
- [x] Verify that pressing anywhere causes the top right Bar Buttons to refresh: Add Checklist and Dismiss Keyboard should show up

### Test: Markdown
1. Access the Notes List
2. Press over any note
3. Enable Markdown Preview
4. Slide from the right to push the Markdown Preview UI

- [x] Verify the top left navigation button allows you to go back

### Test: Blank Notes
1. Access the Notes List
2. Press over the top right `+` icon
3. Don't type anything!
4. Press back

- [x] Verify the newly created note isn't anywhere to be found in the Notes List!

### Release
These changes do not require release notes.
